generator/src/main/java/com/scottlogic/deg/generator/inputs/AtomicConstraintReaderLookup.java

```diff
@@ -271,16 +274,25 @@ private static LocalDateTime unwrapDate(Object value) throws InvalidProfileExcep
         throw new InvalidProfileException(String.format("Dates should be expressed in object format e.g. { \"date\": \"%s\" }", value));
     }
 
-    private static LocalDateTime parseDate(String value) throws InvalidProfileException {
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'.'SSS");
+    static LocalDateTime parseDate(String value) throws InvalidProfileException {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("u-MM-dd'T'HH:mm:ss'.'SSS");
```

Comment:
why change this to `u`?
_first:_ won't this give single digit years instead of `0` padded years ?
_second:_ do we need years before `0000` ?

Reply:
permitting dates before-0001 and after-9999 allows for the testing of this more appropriately.
Otherwise i cannot create a test that asserts that 0000 throws our explicit exception.